CHANGELOG.md

```diff
@@ -3,18 +3,93 @@
 All notable changes to this project will be documented in this file. This
 project adheres to [Semantic Versioning](http://semver.org/).
 
-## Unreleased
+## 0.56.0
+
+This release contains a mix of new features, bugfixes and a new builtin function.
+
+### Support for General References in Rule Heads (Experimental)
+
+A new experimental feature in OPA is support for general refs in rule heads. Where a general ref is a reference with variables at arbitrary locations.
+
+```rego
+package example
+
+import future.keywords
+
+# A partial object rule that converts a list of users to a mapping by "role" and then "id".
```

Comment:
I think ideally, we'd shift terminology we use to single-value and multi-value rules. This, albeit being the more generic cousin of a partial object rule, would be single-value rule. It's pretty much the same as
```rego
users_by_role.admin["id123"] := "alice"
```
except that it happens to have vars instead of strings in the rule (head) ref.

Reply:
You mean as in how we internally differentiate between set-building rules as multi-value, and object-building and primitives as single-value?
There's no need of differentiating between complete- and partial rules? Where the former terminates with an error on assignment conflicts, and the latter is additive to the result on conflicts.

Reply:
Thinking about it a bit more, perhaps the single-/multi-value distinction covers all bases 🤔 ; as key conflicts for partial objects does result in an error too. I'll strip the description of mentions of partial-/complete-; they're not really necessary anyways.